src/layouts/base-layout/index.tsx

```diff
@@ -13,68 +12,44 @@ import usePageviewAnalytics from '@hashicorp/platform-analytics'
 import createConsentManager from '@hashicorp/react-consent-manager/loader'
 
 // Global imports
-import { generateTopLevelSubNavItems } from 'lib/generate-top-level-sub-nav-items'
 import useScrollPercentageAnalytics from 'hooks/use-scroll-percentage-analytics'
 import { CoreDevDotLayoutWithTheme } from 'layouts/core-dev-dot-layout'
 import { CommandBarProvider } from 'components/command-bar'
 import Footer from 'components/footer'
-import MobileMenuContainer, {
-	MobileAuthenticationControls,
-} from 'components/mobile-menu-container'
 import NavigationHeader from 'components/navigation-header'
-import { SidebarNavMenuItem } from 'components/sidebar/components'
 import alertBannerData from 'data/alert-banner.json'
 
 // Local imports
-import { BaseNewLayoutProps, AlertBannerProps } from './types'
-import s from './base-new-layout.module.css'
+import { BaseLayoutProps, AlertBannerProps } from './types'
+import s from './base-layout.module.css'
 
 const { ConsentManager, openConsentManager } = createConsentManager({
 	preset: 'oss',
 })
 
 /**
- * The mobile menu that shows on non-product pages, or pages that do not use the
- * SidebarSidecarLayout that usually handles the mobile menu.
+ * Renders a layout with a navigation header, footer, and content area.
+ *
+ * Note: this layout will _always_ render a `MobileMenuButton`, even if
+ * no `mobileMenuSlot` is provided. When `BaseLayout` is used directly,
+ * we expect the consumer of the layout to render a `MobileMenuContainer`
+ * somewhere else in their view, or provide `mobileMenuSlot` with a
+ * `MobileMenuContainer` to render.
+ *
+ * If you need a layout with a built-in generic global mobile menu,
+ * use `BaseLayoutGenericMobileMenu`.
  */
-const NonProductPageMobileMenu = () => {
-	return (
-		<MobileMenuContainer className={s.mobileMenuContainer}>
-			<MobileAuthenticationControls />
-			<ul className={s.mobileMenuNavList}>
-				<SidebarNavMenuItem item={{ heading: 'Main Menu' }} />
-				{generateTopLevelSubNavItems().map((item: $TSFixMe, index: number) => (
-					// eslint-disable-next-line react/no-array-index-key
-					<SidebarNavMenuItem item={item} key={index} />
-				))}
-			</ul>
-		</MobileMenuContainer>
-	)
-}
-
-/**
- * TODO (future enhancement): rename and abstract `SidebarNavDataProvider` for
- * use here.
- */
-const BaseNewLayout = ({
+const BaseLayout = ({
 	children,
 	showFooterTopBorder = false,
-}: BaseNewLayoutProps) => {
-	const router = useRouter()
-
+	mobileMenuSlot,
```

Comment:
(idea)
What if we removed the `BaseLayoutGenericMobileMenu` wrapper (which feels little like an unnecessary _wrapper-with-default-mobileMenuSlot-prop_) and did either...
1. set a default prop: `mobileMenuSlot={<NonProductPageMobileMenu />}`
2. or require `BaseLayout` consumers to explicitly specify the `mobileMenuSlot`. Acknowledging this is less DRY in practice, but this less-DRY-ness is realistically not that big of a problem.
  - I'm thinking of this as "A little duplication > an incorrect abstraction"

Reply:
☝️ Ooh yeah I dig this direction! Updated to remove `<BaseLayoutGenericMobileMenu>` - replaced all such uses with `<BaseLayout mobileMenuSlot={<MobileMenuLevelsGeneric />}>` 👍

The above change doesn't quite meet `1.` or `2.` exactly - definitely looking forward to improving this further post-API-docs work! Got a task at 🎟 [[discuss] Refactor MobileMenu to remove context, use portal instead of z-index](https://app.asana.com/0/1202097197789424/1205087147421651/f) where I added your notes here cause I think making `mobileMenuSlot` required seems like a great directly 👍